vhdbuilder/scripts/linux/mariner/tool_installs_mariner.sh

```diff
@@ -34,6 +34,25 @@ networkdWorkaround() {
     sed -i "s/Name=e\*/Name=eth0/g" /etc/systemd/network/99-dhcp-en.network
 }
 
+# On Mariner 2.0 Marketplace images, the default systemd network config is removed.
+# Because Azure data sources is disabled in AKS, we need to readd a default configuration for eth0
+setMarinerNetworkdConfig() {
```

Comment:
what's the actual behavior difference between this and networkdWorkaround?

one seds the config in place, one overwrites the whole file? can we collapse?

Reply:
or is there some real difference in the base config?

Reply:
There is a slight difference between 1.0 and 2.0 configs in this case, but I combined them into setMarinerNetworkdConfig and branch inside of the function rather than inside the caller.